brave/brave-core
# Support Ads history and feedback loop

Fixes brave/brave-browser#4047

UI PR: https://github.com/brave/brave-ui/pull/465

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Create a new profile
- Launch Brave and view several ads
- Visit brave://rewards and click 7-day Ads History link in Ads section
- No more than 7 days worth of ads should ever appear in this list
- Click thumbs down on an ad and ensure that:
  - The ad appears in `filtered_ads` section of `client.json` file in user profile directory
  - No additional ads in this creative set appear
- Click thumbs up on the same ad and ensure that:
  - The ad is removed from the `filtered_ads` section of `client.json`
  - This ad and other ads in this creative set should again be eligible for viewing
- Click heart icon for a category and ensure that:
  - All entries with that category are hearted
- Click circle-and-slash (prohibit) icon on the same ad and ensure that:
  - All entries with that category are "prohibited"
  - This category appears in `filtered_categories` section of `client.json`
- Save and Mark as Inappropriate (under ... icon) work similarly:
  - Clicking them should reflect new status
  - Entry should appear in `saved_ads` or `flagged_ads` as appropriate
  - Flagged ads should not be shown, same as thumbs down ads
  - Saving an ad should create a new "toolbar" at the top of the page, allowing you to see the saved ads

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
